<a name="i2b"/>
<h3>i2b</h3>
<p>USAGE: bambi i2b [options]
<p>
<p>Convert Illumina BCL to BAM, SAM, or CRAM file.
<p>
This command will take the output from an Illumina run, the intensities directory, basecalls directory, and so on,
and convert the information into a BAM file (or SAM or CRAM depending on the --output-fmt paramater).
</p>
<p>
Most of the paramaters are read from the various configuration XML files, but may be overwritten on the 
command line.
</p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>--intensity-dir -i</td><td>Illumina intensities directory including config xml file, and clocs, locs or pos files under lane directory.  Required. </td></tr>
<tr><td>--run-folder -r</td><td>Illumina runfolder directory including runParameters xml file under it [default: two levels from Intensities directory]</td></tr>
<tr><td>--basecalls-dir -b</td><td>Illumina basecalls directory including config xml file, and filter files, bcl, maybe scl files under lane cycle directory [default: BaseCalls directory under intensities directory]</td></tr>
<tr><td>--lane -l</td><td>Lane number.  Required. </td></tr>
<tr><td>--output-file -o</td><td>Output file name. May be '-' for stdout. Required. </td></tr>
<tr><td>--generate-secondary-basecalls</td><td>Including second base call or not, [default: false]</td></tr>
<tr><td>--no-filter</td><td>Do not filter cluster [default: false]</td></tr>
<tr><td>--read-group-id</td><td>ID used to link RG header record with RG tag in SAM record, [default: "1"]</td></tr>
<tr><td>--library-name</td><td>The name of the sequenced library [default: 'unknown']</td></tr>
<tr><td>--sample-alias</td><td>The name of the sequenced sample [default: same as library name]</td></tr>
<tr><td>--study-name</td><td>The name of the study [default: none]</td></tr>
<tr><td>--platform-unit</td><td>The platform unit [default: runfolder name plus lane number]</td></tr>
<tr><td>--run-start-date</td><td>The start date of the run in ISO8610 format [default: read from config file]</td></tr>
<tr><td>--sequencing-centre</td><td>Sequence center name [default: 'SC']</td></tr>
<tr><td>--platform</td><td>The name of the sequencing technology that produced the reads [default: 'ILLUMINA']</td></tr>
<tr><td>--first-tile</td><td>If set, this is the first tile to be processed. Normally used for debugging and testing. [default: none]</td></tr>
<tr><td>--tile-limit</td><td>If set, process no more than this many tiles. Normally used for debugging and testing.  [default: all tiles]</td></tr>
<tr><td>--barcode-tag</td><td>Comma separated list of tag names for barcode sequences. [default: BC]</td></tr>
<tr><td>--quality-tag</td><td>Comma separated list of tag name for barcode qualities. [default: QT]</td></tr>
<tr><td>--bc-read</td><td>Comma separated list of Which reads (1 or 2) should the barcode sequences and qualities be added to [default: '1']</td></tr>
<tr><td>--first-cycle</td><td>First cycle for each standard (non-index) read. Comma separated list</td></tr>
<tr><td>--final-cycle</td><td>Last cycle for each standard (non-index) read. Comma separated list.</td></tr>
<tr><td>--first-index-cycle</td><td>First cycle for each index read. Comma separated list.</td></tr>
<tr><td>--final-index-cycle</td><td>Last cycle for each index read. Comma separated list.</td></tr>
<tr><td>--no-index-separator -s</td><td>Do NOT separate multiple indexes with a '-' character, just concatenate instead. [default: false]</td></tr>
<tr><td>--verbose -v</td><td>Display progress messages</td></tr>
<tr><td>--output-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--compression-level</td><td>[0..9]</td></tr>
</table>
<br/>

<a name="select"/>
<h3>select</h3>
<p>USAGE: bambi select [options]
<p>
<p>Give a list of SAM/BAM files with the same set of records and in the same order but aligned with different references, split reads into different files according to alignments. You have option to put unaligned reads into one of output files or a separate file.
<p></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>--input -i</td><td>Comma seperated list of input files</td></tr>
<tr><td>--output -o</td><td>Comma seperated list of output files</td></tr>
<tr><td>-n</td><td>The output SAM or BAM file for unaligned reads. If not given, the unaligned reads will put into the last output file.</td></tr>
<tr><td>-m</td><td>Metrics file name  Default value: null. </td></tr>
<tr><td>--verbose -v</td><td>Display progress messages</td></tr>
<tr><td>--input-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--output-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--compression-level</td><td>[0..9]</td></tr>
</table>
<br/>

<a name="decode"/>
<h3>decode</h3>
<p>USAGE: bambi decode [options] <filename>
<p>
<p>A command-line tool to decode a multiplexed bam file.
<p></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>--output -o</td><td>The output file after decoding. [default: stdout]</td></tr>
<tr><td>--barcode-file -b</td><td>Tab-delimited file of barcode sequences, and optionally barcode name and library name.  Barcodes must be unique, and all the same length.  Column headers must be 'barcode_sequence', 'barcode_name', and 'library_name'.  Required.</td></tr>
<tr><td>--metrics-file</td><td>Per-barcode and per-lane metrics written to this file.  Required. </td></tr>
<tr><td>--convert-low-quality</td><td>Convert low quality bases in barcode reads to 'N' [default: false]</td></tr>
<tr><td>--max-low-quality-to-convert</td><td>Max low quality phred value to convert bases in barcode read to 'N' [default: 15]</td></tr>
<tr><td>--max-no-calls</td><td>Max allowable number of no-calls in a barcode read before it is considered unmatchable [default: 2]</td></tr>
<tr><td>--max-mismatches</td><td>Maximum mismatches for a barcode to be considered a match [default: 1]</td></tr>
<tr><td>--min-mismatch-delta</td><td>Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match [default: 1]</td></tr>
<tr><td>--change-read-name</td><td>Change the read name by adding #<barcode> suffix [default: do not change]</td></tr>
<tr><td>--barcode-tag-name</td><td>Barcode tag name [default: BC]</td></tr>
<tr><td>--quality-tag-name</td><td>Quality tag name [default: QT]</td></tr>
<tr><td>--dual-tag</td><td>Dual tag position in the barcode string (between 2 and barcode length - 1). Only required if the dual tags are not separated by a '-'</td></tr>
<tr><td>--verbose -v</td><td>Display progress messages</td></tr>
<tr><td>--input-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--output-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--compression-level</td><td>[0..9]</td></tr>
</table>
<br/>

<a name="chrsplit"/>
<h3>chrsplit</h3>
<p>USAGE: bambi chrsplit [options]
<p>
<p>Split a BAM (or SAM) file into two files; one for records matching given chromosome subsets (target), and one for all other records (excluded). Original headers are preserved, with additional @PG entry.
<p></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>--input -i</td><td>Input SAM or BAM file  Required. </td></tr>
<tr><td>--output -o</td><td>Output SAM/BAM path for target sequences.  Required. </td></tr>
<tr><td>--exclude -e</td><td>Output SAM/BAM path for excluded sequences.  Required. </td></tr>
<tr><td>--subset -s</td><td>Comma separated list of references to target [default: 'MT,Y']</td></tr>
<tr><td>--exclude-unaligned -u</td><td>Exclude read groups in which all reads are unaligned (Groups with at least one read aligned to the target, and others unaligned, will be written to the target file.)</td></tr>
<tr><td>--invert -V</td><td>Treat the S option as a list to EXCLUDE rather than TARGET, so that chimeric/unmapped read pairs remain excluded.</td></tr>
<tr><td>--verbose -v</td><td>Display progress messages</td></tr>
<tr><td>--input-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--output-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--compression-level</td><td>[0..9]</td></tr>
</table>
<br/>

<a name="read2tags"/>
<h3>read2tags</h3>
<p>USAGE: bambi read2tags [options]
<p>
<p>Convert a portion of the reads for each record into aux tags.</p>
<p>If all of the read is used up, then it is merged with it's paired record.</p>
<p></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>--input -i</td><td>Input SAM or BAM file  Required. </td></tr>
<tr><td>--output -o</td><td>Output SAM/BAM path for target sequences.  Required. </td></tr>
<tr><td>--tags -t</td><td>Comma separated list of barcode tags</td></tr>
<tr><td>--qtags -q</td><td>Comma separated list of quality tags</td></tr>
<tr><td>--keep-tags -k</td><td>Comma separated list of tags to keep when merging records [default: BC,QT,RG]</td></tr>
<tr><td>--discard-tags -k</td><td>Comma separated list of tags to discard when merging records [default: as,af,aa,a3,ah]</td></tr>
<tr><td>--positions -p</td>
<td>Comma separated list of positions within the read to convert<br/>
The format is r:s:e,r:s:e,r:s:e,...<br/>
r is record number (0, 1 or 2 and is optional)<br/>
s is the start position in the read string<br/>
e is the end position in the read string<br/>
Note that the start and end positions are 1 (not zero) based<br/>
</td></tr>
<tr><td>--merge -m</td><td>Merge duplicate tags</td></tr>
<tr><td>--replace -r</td><td>Replace duplicate tags</td></tr>
<tr><td>--verbose -v</td><td>Display progress messages</td></tr>
<tr><td>--input-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--output-fmt</td><td>[sam/bam/cram] [default: 'bam']</td></tr>
<tr><td>--compression-level</td><td>[0..9]</td></tr>
</table>
<br/>

